openmc/deplete/microxs.py

```diff
@@ -67,26 +66,36 @@ def from_model(cls,
             Cross section data in [b]
 
         """
-        groups = EnergyGroups(energy_bounds)
-
         # Set up the reaction tallies
         original_tallies = model.tallies
         original_materials = deepcopy(model.materials)
-        tallies = Tallies()
         xs = {}
-        reactions, diluted_materials = cls._add_dilute_nuclides(chain_file,
-                                                                model,
-                                                                dilute_initial)
+        reactions, burnable_nucs, diluted_materials = cls._add_dilute_nuclides(
+            chain_file, model, dilute_initial)
         model.materials = diluted_materials
 
-        for rx in reactions:
-            if rx == 'fission':
-                xs[rx] = FissionXS(domain=reaction_domain,
-                                   energy_groups=groups, by_nuclide=True)
-            else:
-                xs[rx] = ArbitraryXS(rx, domain=reaction_domain,
-                                     energy_groups=groups, by_nuclide=True)
-            tallies += xs[rx].tallies.values()
+        energy_filter = openmc.EnergyFilter(energy_bounds)
+        if isinstance(reaction_domain, openmc.Material):
+            domain_filter = openmc.MaterialFilter([reaction_domain])
+        elif isinstance(reaction_domain, openmc.Cell):
+            domain_filter = openmc.CellFilter([reaction_domain])
+        elif isinstance(reaction_domain, openmc.Universe):
+            domain_filter = openmc.UniverseFilter([reaction_domain])
+        else:
+            raise ValueError(f"Unsupport domain type: {type(reaction_domain)}")
+
+        # TODO: Right now, we use all nuclides from the material but it probably
+        # should be based on the burnable nuclides
```

Comment:
Agreed! It won't be much use to tally rates for a nuclide that has no cross sections, or isn't in the depletion chain. I think in practice, `MicroXS` behaves this way anyways, so I think you should be able to make this change in this PR.

Reply:
I thought about doing it here but I decided to defer that for the next PR, which will also remove `dilute_initial` and result in test result changes. I wanted this PR to be focused solely on refactoring how tallies are setup in `MicroXS.from_model` while preserving the existing test results.